src/Storages/MergeTree/ReplicatedMergeTreeLogEntry.h

```diff
@@ -174,7 +173,36 @@ struct ReplicatedMergeTreeLogEntryData
     size_t quorum = 0;
 
     /// Used only in tests for permanent fault injection for particular queue entry.
-    bool fault_injected = false;
+    struct CopyableAtomicFlag
```

Comment:
May be worth to move to Common, I remember I did have similar problem and this struct could be handy

Reply:
AFAIK the constructor is forbidden because they want to allow you to be explicit with memory ordering during copy.
But I guess the default ordering should be safe enough. I'll move it to Common.